util/src/main/java/edu/asu/diging/gilesecosystem/util/files/impl/FileStorageManager.java

```diff
@@ -237,6 +237,38 @@ public String getBaseDirectory() {
     public String getBaseDirectoryWithFiletype() {
         return baseDirectory + File.separator + fileTypeFolder;
     }
+    
+    @Override
+    public byte[] getExtractedFileContent(String username, String uploadId, String documentId, int pageNr, String filename) {
+        String folderPath = getAndCreateStoragePath(username, uploadId, documentId);
+        File fileObject = new File(folderPath + File.separator + "extracted" + File.separator + pageNr + File.separator + "extracted" + File.separator + filename);
+        try {
+            return getFileContentFromUrl(fileObject.toURI().toURL());
+        } catch (IOException e) {
+            logger.error("Could not read file.", e);
+            return null;
```

Comment:
you probably want to throw an exception here since this should not happen under normal circumstances.